### PR TITLE
Blazemod update

### PR DESCRIPTION
**Pull request by chaosvolt**
http://smf.cataclysmdda.com/index.php?topic=8060.msg234106#msg234106

> By all means, if someone could update it, please do.

Copypasta from the latest version of Vehicle Additions Pack from the
forum thread, should fix some turret install and/or load errors currently
present.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
The `"fuel_type"` tag is outdated on weapons. This will require a big search+replace before merging.
Preferably on Blaze's side, so that he unclutters his mod for future updates (if he wants to keep updating it).

File: data/mods/blazemod/blaze_weapons_parts.json [hunk 1]
```diff
@@ -115,6 +120,7 @@
         "broken_color" : "dark_gray",
         "damage_modifier" : 80,
         "durability" : 400,
+        "fuel_type" : "50",
```
Comment (BevapDin) [inline data/mods/blazemod/blaze_weapons_parts.json]: A note about this: the game takes the ammo type now directly from the gun of the turret, putting it here as fuel type is not needed anymore. It is actually wrong because fuel types are item types, not ammo types (see #12411 for the reaching about this).

And a final note  (you may have already done this), please go into the vehicle info and try to install a part and scroll through *all* listed item. This should not not give any debug messages. Listing an ammo type (which is not by chance also an item type) here will show a debug message.

In short: just remove the fuel_type entry from all parts with the TURRET flag (it is only required for parts with the FUEL_TANK or the ENGINE flag)

**Comment by chaosvolt** [comment]
Blarghle. What's the equivalent tag now? Not sure whether to peaster Blaze to fix that or do it here myself. >w<

File: data/mods/blazemod/blaze_override.json [hunk 2]
```diff
@@ -258,5 +258,24 @@
             {"item": "steel_chunk", "min": 5, "max": 10},
             {"item": "scrap", "min": 5, "max": 10}
         ]
+    },{
```
Comment (Coolthulhu) [inline data/mods/blazemod/blaze_override.json]: Does this entry actually change anything from the usual battery mount? If not, it should be removed.

File: data/mods/blazemod/blaze_autoweapons.json [hunk 3]
```diff
@@ -41,7 +41,7 @@
     "price": 280000,
     "material": ["steel", "plastic"],
 	"ammo_effects": ["BOUNCE", "LIGHTNING", "MOUNTED_GUN"],
-    "flags": ["NO_AMMO", "NEVER_JAMS",  "NO_UNLOAD", "USE_UPS_200"],
+    "flags": ["NEVER_JAMS",  "NO_UNLOAD", "USE_UPS_200"],
```
Comment (BevapDin) [inline data/mods/blazemod/blaze_autoweapons.json]: The "USE_UPS_*" flags don't do anything anymore. The used UPS charges are defined directly as a member:
```JSON
{
    "id": "ateslagun",
    "ups_charges": 200,
```

Hmm, seems I haven't properly updated that in all the mods when I changed it, I apologize.

**Comment by chaosvolt** [comment]
I'm not entirely sure, but note that Blaze did upload an update to Mediafire. Gonna get it and commit it here, then see if it changed these added things.

**Comment by tyrael93** [comment]
Are rocks still classified as ammo in this version?

**Comment by chaosvolt** [comment]
> Are rocks still classified as ammo in this version?

Seems to be the case, yes. The derpiness that causes has been discussed in an issue I opened, unsure if its flammability has been corrected. o3o

**Comment by DavidKeaton** [comment]
glad this made it in!

**Comment by chaosvolt** [comment]
Yus. Thank you Coolthulhu for the merge, and thanks to Blaze for letting me PR it. owo
